### ISSUE TITLE
Unstable test KerberosStandaloneCrossRealmTrustTest.test03SpnegoLoginWithCorrectKerberosPrincipalRealm

### ISSUE THREAD

**Issue by ahus1** [keycloak/keycloak#30037]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

The test `KerberosStandaloneCrossRealmTrustTest.test03SpnegoLoginWithCorrectKerberosPrincipalRealm` failed in 

https://github.com/keycloak/keycloak/actions/runs/9317327510/job/25647680793?pr=30027

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

Test should succeed

### Actual behavior

Test randomly fails.

### How to Reproduce?

See job above

**Comment by mposolda** [comment]
@ahus1 We are closing this as we did not see this issue anymore on the latest dashboard https://stianst.github.io/keycloak-dashboard/tests . Feel free to re-open if you still see it (or bot will possibly open new issue if it is still valid).

**Comment by stianst** [comment]
Failed again here: https://github.com/stianst/keycloak/actions/runs/11970101528/job/33372329062

**Comment by rmartinc** [comment]
I have executed the test `KerberosStandaloneCrossRealmTrustTest` 50 times in my laptop and in the CI ([here](https://github.com/rmartinc/keycloak/actions/runs/12006920697/job/33466764173)) with a for and it didn't fail for me. So this is very complicated to get a fix for this with such rate of failures. My feeling is that something is not still initialized (ldap, kerberos or similar).

**Comment by rmartinc** [comment]
This is never failing for me. No matter what I do. I have now executed all the kerberos tests `-Dtest=org.keycloak.testsuite.federation.kerberos.**.*Test` 50 times in this [run](https://github.com/rmartinc/keycloak/actions/runs/12065371653). No error. 

The log of the failed test is the following:

```
2024-11-22T11:27:40.1296555Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,136 INFO  [org.keycloak.testsuite.federation.kerberos.KerberosStandaloneCrossRealmTrustTest] [KerberosStandaloneCrossRealmTrustTest] test03SpnegoLoginWithCorrectKerberosPrincipalRealm() STARTED
2024-11-22T11:27:40.1298728Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,275 INFO  [org.keycloak.testsuite.util.URLUtils] starting navigation
2024-11-22T11:27:40.1300265Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,275 INFO  [org.keycloak.testsuite.util.URLUtils] current URL:  https://localhost:8543/auth/realms/master/app/auth
2024-11-22T11:27:40.1302072Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,275 INFO  [org.keycloak.testsuite.util.URLUtils] navigating to https://localhost:8543/auth/realms/master/testing/blank
2024-11-22T11:27:40.1303915Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,281 INFO  [org.keycloak.testsuite.util.URLUtils] new current URL:  https://localhost:8543/auth/realms/master/testing/blank
2024-11-22T11:27:40.1305485Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,282 INFO  [org.keycloak.testsuite.util.URLUtils] navigation complete
2024-11-22T11:27:40.1307199Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,282 INFO  [org.keycloak.testsuite.federation.kerberos.KerberosStandaloneCrossRealmTrustTest] deleting cookies in 'master' realm
2024-11-22T11:27:40.1309932Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,283 INFO  [org.keycloak.testsuite.util.KerberosRule] Krb5.conf file location is: /home/runner/work/keycloak/keycloak/testsuite/integration-arquillian/tests/base/target/test-classes/kerberos/test-krb5.conf
2024-11-22T11:27:40.1312484Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,513 WARN  [org.apache.directory.server.protocol.shared.kerberos.StoreUtils] No server entry found for kerberos principal name jduke@KC2.COM
2024-11-22T11:27:40.1314462Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,513 WARN  [org.apache.directory.server.KERBEROS_LOG] No server entry found for kerberos principal name jduke@KC2.COM
2024-11-22T11:27:40.1316340Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,513 ERROR [org.apache.directory.server.KERBEROS_LOG] Error while searching for client jduke@KC2.COM : Client not found in Kerberos database
2024-11-22T11:27:40.1318345Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,513 WARN  [org.apache.directory.server.kerberos.protocol.KerberosProtocolHandler] Client not found in Kerberos database (6)
2024-11-22T11:27:40.1320127Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:29,513 WARN  [org.apache.directory.server.KERBEROS_LOG] Client not found in Kerberos database (6)
2024-11-22T11:27:40.1321891Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:39,517 INFO  [org.keycloak.testsuite.federation.kerberos.KerberosStandaloneCrossRealmTrustTest] calling all TestCleanup
2024-11-22T11:27:40.1324704Z KerberosStandaloneCrossRealmTrustTest ++ 11:27:39,545 ERROR [org.keycloak.testsuite.federation.kerberos.KerberosStandaloneCrossRealmTrustTest] [KerberosStandaloneCrossRealmTrustTest] test03SpnegoLoginWithCorrectKerberosPrincipalRealm() FAILED
```

So the important part is this line:

```
[org.apache.directory.server.protocol.shared.kerberos.StoreUtils] No server entry found for kerberos principal name jduke@KC2.COM
```

This is supposed that is searching a user with filter `krb5PrincipalName=jduke@KC2.COM` and it's not found by the ldap itself. But the user is defined [here](https://github.com/keycloak/keycloak/blob/26.0.6/testsuite/integration-arquillian/tests/base/src/test/resources/kerberos/users-kerberos-kc2.ldif#L104). So it should be there. I don't know why the ldap is not finding the user. And I cannot make it fail.

**Comment by rmartinc** [comment]
We agreed to close this one as we cannot reproduce and the logs show the ldap server is no finding the user via kerberos pricipal attribute. Maybe we should re-open it later but we cannot reproduce it.

**Comment by stianst** [comment]
https://github.com/stianst/keycloak/actions/runs/12237171414

**Comment by rmartinc** [comment]
@stianst I don't know what you want us to do with this one. I have done a few changes (but I don't think I'm doing much) and I'm executing without retry 100 times. I have changed to execute 10 base5 CI execution and I executed manually 10 times (100 in total, I'm now in  the 10th run here https://github.com/rmartinc/keycloak/actions/runs/12257118055). And it's not failing. This can be anything (apacheds, kerberos apacheds, something in the CI env, even jdk,...). It may fail one in 5 hundred times? It's impossible to be sure if it is fixed or even if the issue is in our code. As commented, I have done a few changes but we will never be sure if we fix this and, even worse, we don't know how to fix it because it's hardly reproducible.

I will send the few changes I have done, but it's just a blind shot.
